src/onelogin/saml2/idp_metadata_parser.py

```diff
@@ -108,9 +126,12 @@ def parse(idp_metadata):
 
                     if idp_entity_id is not None:
                         data['idp']['entityId'] = idp_entity_id
+
                     if idp_sso_url is not None:
                         data['idp']['singleSignOnService'] = {}
                         data['idp']['singleSignOnService']['url'] = idp_sso_url
+                        data['idp']['singleSignOnService']['binding'] = preferred_sso_binding
```

Comment:
You set here the preferred_sso_binding but maybe it was extracted the first singleSignOnService endpoint that has a different binding